### PR TITLE
paddingNewLinesAfterBlocks: ease up on function calls

### LINKED ISSUE #1232: requirePaddingNewLinesAfterBlocks seems overly strict
```js
callingAFunction(
  with,
  another,
  function() {
    // ...
  }
);

// more code
```

This results in `Missing newline after closing curly brace`. IMO the function call is a block in itself, so the newline should be required after it. Not after the nested anonymous function.

But JSCS prefers this – which should IMHO be wrong:

```js
callingAFunction(
  with,
  another,
  function() {
    // ...
  }

);
// more code
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/10348543-requirepaddingnewlinesafterblocks-seems-overly-strict?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by markelog**
Fixes #1232

### CODE REVIEW

**Comment by mikesherov** [comment]
What about function calls that aren't the last parameter?

**Comment by hzoo** [comment]
@mikesherov I was suggesting check `nextToken.value === ','` as well if that works? I think that is the issue for #1237.

**Comment by markelog** [comment]
@hzoo nope, have an idea, checking it out

**Comment by markelog** [comment]
Okay, how about now?

**Comment by hzoo** [comment]
Yeah #1237 passes with that.

**Comment by markelog** [comment]
Oh, cool :-)

File: lib/rules/require-padding-newlines-after-blocks.js [hunk 1]
```diff
@@ -105,6 +105,15 @@ module.exports.prototype = {
                     return;
                 }
 
+                // Check if this block is used in call expression
+                if (parentNode.parentNode.type === 'CallExpression') {
```
Comment (mikesherov) [inline lib/rules/require-padding-newlines-after-blocks.js]: much better. Finally, using the ast as expected. ;-)

**Comment by mikesherov** [comment]
:+1:

**Comment by mikesherov** [comment]
@TheSavior 's comment still applies.

**Comment by TheSavior** [comment]
I still don't think this is the right solution as I don't believe this passes the tests that I added in my solution. @mikesherov please look at @1239 for comparison.

File: lib/rules/require-padding-newlines-after-blocks.js (hunk 1 above)
Comment (zxqfox) [inline lib/rules/require-padding-newlines-after-blocks.js]: According to @TheSavior comment: feels like we need to add an exception for `Program` node. No?

Comment (TheSavior) [inline lib/rules/require-padding-newlines-after-blocks.js]: See https://github.com/jscs-dev/node-jscs/pull/1239/files#diff-3c958cdc2aa18d8507669afbc4372ee5R154

Comment (mikesherov) [inline lib/rules/require-padding-newlines-after-blocks.js]: @TheSavior 's solution is correct in that it won't barf on top level blocks. However, I don't think we need to add any options.

Comment (TheSavior) [inline lib/rules/require-padding-newlines-after-blocks.js]: My solution also handles arrays, I'm happy to remove the configuration options.

Comment (markelog) [inline lib/rules/require-padding-newlines-after-blocks.js]: So test-case, code-example?

> My solution also handles arrays

I'm not sure if we need that at the moment

Comment (TheSavior) [inline lib/rules/require-padding-newlines-after-blocks.js]: @markelog Can you make a case why we wouldn't want this to be handled in arrays?

Comment (markelog) [inline lib/rules/require-padding-newlines-after-blocks.js]: We require newline char in objects but not in arrays? What is the diff?

Comment (markelog) [inline lib/rules/require-padding-newlines-after-blocks.js]: Alrighty, i don't see an example when we would need exception for `parentNode.parentNode` top-level element. 

If find one, let me know

**Pull request closed without merging**